File: Search/c++/15_부품찾기(이진탐색).cpp
// 이것이 취업을 위한 코딩테스트다 197p
// 1

#include <iostream>
#include <vector>

using namespace std;

int N, M;

int binary_search(vector<int> arr, int target, int start, int end) {
	if (start > end)
		return -1;
	int mid = (start + end) / 2;
	if (arr[mid] == target)
		return arr[mid];
	else if (arr[mid] < target)
		return binary_search(arr, target, mid + 1, end);
	else
		return binary_search(arr, target, start, end - 1);
}

int main() {
	cin >> N;
	vector<int> arr(N);
	for (int i = 0; i < N; i++)
		cin >> arr[i];
	cin >> M;
	vector<int> chk(M);
	for (int i = 0; i < M; i++)
		cin >> chk[i];

	for (auto i : chk) {
		if (binary_search(arr, i, 0, N - 1) != -1)
			cout << "yes ";
		else
			cout << "no ";
	}
}

File: Search/python/15_부품찾기(이진탐색).py
# 이것이 취업을 위한 코딩테스트다 197p
# 1

def binary_search(arr, target, start, end):
    while start <= end:
        mid = (start + end) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] > target:
            end = mid - 1
        else:
            start = mid + 1
    return None

N = int(input())
arr = list(map(int, input().split()))
arr.sort()
M = int(input())
tmp_list = list(map(int, input().split()))

for i in tmp_list:
    result = binary_search(arr, i, 0, N - 1)
    if result != None:
        print("yes", end=" ")
    else:
        print("no", end=" ")